lib/datasets/single_file_imdb.py

```diff
@@ -0,0 +1,153 @@
+        # self._roidb_handler = self.gt_roidb
+        self.competition_mode(False)
+
+    def parse_text_dataset(self):
+        db_path = os.path.join(self._data_path, self._image_set_txt)
+        with open(db_path, 'r') as f:
+            image_filenames = []
+            image_gt_boxes = []
+            gt_boxes = []
+
+            for line_idx, line in enumerate(f.readlines()):
+                if len(line) == 0:
+                    # ignore empty lines
+                    continue
+                if line[0] == ' ':
+                    # this line describes a bounding box for the previous image
+                    if len(image_filenames) == 0:
+                        raise ValueError('found bbox entries before first image path on line %d' % (line_idx + 1))
+                    line = line.replace(',', '')
+                    splitLine = line.split()
+                    cls = splitLine[4]
+                    left, top, right, bottom = map(float, splitLine[:4])
+                    gt_boxes.append((left, top, right, bottom, cls))
+                else:
+                    # this line describes an image
+                    if len(image_filenames) > 0:
+                        # finalize the ground-truth box collection for the last image
+                        image_gt_boxes.append(gt_boxes)
+                        gt_boxes = []
+                    line = line.strip()
+                    image_filenames.append(os.path.join(self._data_path, line))
+
+            if len(gt_boxes) > 0:
+                # After the end of the file, append the last collection of boxes
+                image_gt_boxes.append(gt_boxes)
+
+        return image_filenames, image_gt_boxes
+
+    def image_path_at(self, i):
+        """
+        Return the absolute path to image i in the image sequence.
+        """
+        return self.image_index[i]
+
+    def gt_roidb(self):
+        """
+        Return the database of ground-truth regions of interest.
+
+        This function loads/saves from/to a cache file to speed up future
+        calls.
+        """
+        cache_file = os.path.join(self.cache_path, self.name + '_gt_roidb.pkl')
+        if os.path.exists(cache_file):
+            with open(cache_file, 'rb') as fid:
+                pklRoidb = cPickle.load(fid)
+            if pklRoidb[0]['gt_overlaps'].shape[1] == self.num_classes:
+                print '{} gt roidb loaded from {}'.format(self.name, cache_file)
+                return pklRoidb
+            else:
+                print 'roidb loaded from {} had wrong class count. Ignoring.'\
+                    .format(cache_file)
+
+        # The roidb we build in __init__ is the ground truth roidb.
+        gt_roidb = self.roidb
+
+        with open(cache_file, 'wb') as fid:
+            cPickle.dump(gt_roidb, fid, cPickle.HIGHEST_PROTOCOL)
+        print 'wrote gt roidb to {}'.format(cache_file)
+        return gt_roidb
+
+    def evaluate_detections(self, all_boxes, output_dir):
```

Comment:
This function is duplicated in `datasets/custom_voc_structure_db.py`. Ideally, it would shared between the classes somehow.